neuralmonkey/decoders/beam_search_decoder.py

```diff
@@ -181,9 +190,22 @@ def cond(*args) -> tf.Tensor:
         dec_loop_state = final_state.decoder_loop_state
         bs_state = final_state.bs_state
 
-        scores = final_state.bs_output.scores.stack()
-        parent_ids = final_state.bs_output.parent_ids.stack()
-        token_ids = final_state.bs_output.token_ids.stack()
+        # Because the "batch" dymension is dynamic and the TensorArrays
+        # are empty during initial call during ensembling we have
+        # to define a default values (the runner itself has to handle them,
+        # if needed
+        scores = tf.cond(
+            tf.equal(final_state.bs_output.scores.size(), 0),
+            lambda: tf.fill([1, self.batch_size, self.beam_size], 0.0),
+            final_state.bs_output.scores.stack)
+        parent_ids = tf.cond(
+            tf.equal(final_state.bs_output.parent_ids.size(), 0),
+            lambda: tf.fill([1, self.batch_size, self.beam_size], 0),
+            final_state.bs_output.parent_ids.stack)
+        token_ids = tf.cond(
+            tf.equal(final_state.bs_output.token_ids.size(), 0),
+            lambda: tf.fill([1, self.batch_size, self.beam_size], 0),
+            final_state.bs_output.token_ids.stack)
```

Comment:
udělal bych to možná jako jeden cond, kterej vrací trojici `scores, parent_ids, token_ids`.

Reply:
když ty podminky jsou furt stejny

Reply:
Pak to ale budu vracet jako tuple a budu muset pridat lambda a vracet primo navratovou hodnotu funkce stack a ne tu fci. To si myslim neni uplne feasible.

Reply:
souhlasim